lib/puppet/type/netscaler_rewriteaction.rb

```diff
@@ -0,0 +1,144 @@
+require 'puppet/property/netscaler_traffic_domain'
+
+Puppet::Type.newtype(:netscaler_rewriteaction) do
+  @doc = 'Manage basic netscaler rewrite action objects.'
+
+  apply_to_device
+  ensurable
+
+  newparam(:name, :parent => Puppet::Parameter::NetscalerName, :namevar => true)
+  newproperty(:type) do
+    desc "Type of rewrite action. It can be: (replace|insert_http_header|delete_http_header|corrupt_http_header|insert_before|insert_after|delete|replace_http_res).
+    For each action type the <target> and <string builder expr> are defined below.
+
+      o insert_http_header: Will insert a HTTP header.
+        <target_expression> = header name.
+        <content_expression> = header value specified as a compound text expression.
+
+      o insert_sip_header: Will insert a SIP header.
+        <target_expression> = header name.
+        <content_expression> = header value specified as a compound text expression.
+
+      o delete_http_header: Will delete all occurrence of HTTP header.
+        <target_expression> = header name.
+
+      o delete_sip_header: Will delete all occurrence of SIP header.
+        <target_expression> = header name.
+
+      o corrupt_http_header: Will corrupt all occurrence of HTTP header.
+        <target_expression> = header name.
+
+      o corrupt_sip_header: Will corrupt all occurrence of SIP header.
+        <target_expression> = header name.
+
+      o replace: Will replace the target text reference with the value specified in attr.
+        <target_expression> = Advanced text expression
+        <content_expression> = Compound text expression
+
+      o insert_before: Will insert the value specified by attr before the target text reference.
+        <target_expression> = Advanced text expression
+        <content_expression> = Compound text expression
+
+      o insert_after: Will insert the value specified by attr after the target text reference.
+        <target_expression> = Advanced text expression
+        <content_expression> = Compound text expression
+
+      o delete: Delete the target text reference.
+        <target_expression> = Advanced text expression
+
+      o replace_http_res: Replace the http response with value specified in target.
+        <target_expression> = Compound text expression
+
+      o replace_sip_res: Replace the SIP response with value specified in target.
+        <target_expression> = Compound text expression
+
+      o replace_all: Replaces all occurrence of the pattern in the text provided in the target with the text provided in the stringBuilderExpr, with a string defined in the -pattern argument or -search argument.
+        For example, you can replace all occurences of abcd with -pattern efgh.
+        <target_expression> = text in a request or a response, for example http.req.body(1000)
+        <content_expression> = Compound text expression
+        -pattern <expression> = string constant, for example -pattern efgh or -search text(\"efgh\")
```

Comment:
Why is this one hyphenated and spaced? Perhaps the docs should be more puppety with parameters.